src/posix/platform/uart.cpp

```diff
@@ -238,7 +238,7 @@ static void InitializeSessionSocket(void)
     // to simply ignore it.
 #if defined(SO_NOSIGPIPE)
     rval = 1;
-    VerifyOrExit((rval = setsockopt(sSessionSocket, SOL_SOCKET, SO_NOSIGPIPE, &rval, sizeof(rval)) != -1, OT_NOOP));
+    VerifyOrExit(rval = setsockopt(sSessionSocket, SOL_SOCKET, SO_NOSIGPIPE, &rval, sizeof(rval)) != -1, OT_NOOP);
```

Comment:
Can the original code compile?

Reply:
The error is revealed when I tried to compile otbr on my macOS catalina.